src/moa/moa_controllers/moa_controllers/ack_to_motec.py

```diff
@@ -22,7 +22,7 @@ def __init__(self):
 
         #create publisher for diagnostics
         self.diagnostics = self.create_publisher(
-            DiagnosticStatus,          #msg type
+            DiagnosticArray,           #msg type
             "diagnostics",             #topic sent to
             10                         #qos
         )
```

Comment:
Please read this: https://github.com/ros/diagnostics/tree/ros2/diagnostic_updater